presto-array/src/main/java/com/facebook/presto/array/ReferenceCountMap.java

```diff
@@ -59,19 +64,44 @@ public long sizeOf()
         return INSTANCE_SIZE + SizeOf.sizeOf(key) + SizeOf.sizeOf(value) + SizeOf.sizeOf(used);
     }
 
-    private static final class ObjectStrategy
-            implements Strategy<Object>
+    /**
+     * Get the 64-bit hash code for an object
+     */
+    private static long getHashCode(Object key)
     {
-        @Override
-        public int hashCode(Object object)
-        {
-            return System.identityHashCode(object);
+        // identityHashCode does not guarantee the uniqueness of an object. We need some extra identity.
```

Comment:
replace `We need some extra identity` with `Any additional identity information can reduce collisions`